### PR TITLE
Feature/apps v2

### PR DESCRIPTION
**Pull request by katherinebernstein**

### CODE REVIEW

File: src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java [hunk 1]
```diff
@@ -20,32 +20,63 @@
...
 
     @Override public AppTypeSetting read( ObjectDataInput in ) throws IOException {
-        UUID entitySetId = UUIDStreamSerializer.deserialize( in );
-        EnumSet<Permission> permissions = DelegatedPermissionEnumSetStreamSerializer.deserialize( in );
-        return new AppTypeSetting( entitySetId, permissions );
+        UUID id = UUIDStreamSerializer.deserialize( in );
+        UUID entitySetCollectionId = UUIDStreamSerializer.deserialize( in );
+
+        int roleMapSize = in.readInt();
+        Map<UUID, AclKey> roleMap = new HashMap<>( roleMapSize );
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java]: Is there a significant difference between `Maps.newHashMapWithExpectedSize( x )` vs `new HashMap<>( x )`?

[the remark this answers is not in the document]
Comment (makosblade) [reply src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java]: It uses google's sizing heuristics which (I think) are better than the default ones in Java?

[the remark this answers is not in the document]
Comment (katherinebernstein) [reply src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java]: That makes me sad but is also understandable.

File: src/test/kotlin/com/openlattice/hazelcast/serializers/UpdateDefaultAppSettingsProcessorStreamSerializerTest.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.openlattice.hazelcast.serializers
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply src/test/kotlin/com/openlattice/hazelcast/serializers/UpdateDefaultAppSettingsProcessorStreamSerializerTest.kt]: :yay:

File: src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java [hunk 3]
```diff
@@ -20,32 +20,63 @@
...
-        DelegatedPermissionEnumSetStreamSerializer.serialize( out, object.getPermissions() );
+        UUIDStreamSerializer.serialize( out, object.getId() );
+        UUIDStreamSerializer.serialize( out, object.getEntitySetCollectionId() );
+
+        out.writeInt( object.getRoles().size() );
+        for ( Map.Entry<UUID, AclKey> entry : object.getRoles().entrySet() ) {
+            UUIDStreamSerializer.serialize( out, entry.getKey() );
+            AclKeyStreamSerializer.serialize( out, entry.getValue() );
+        }
+        out.writeByteArray( mapper.writeValueAsBytes( object.getSettings() ) );
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java]: Is there? I know of the kotlin way...but only of the kotlin way...

File: src/main/java/com/openlattice/postgres/ResultSetAdapters.java [hunk 4]
```diff
@@ -704,33 +707,39 @@ public static PrincipalSet principalSet( ResultSet rs ) throws SQLException {
...
-    public static App app( ResultSet rs ) throws SQLException {
-        UUID id = id( rs );
-        String name = name( rs );
-        String title = title( rs );
-        Optional<String> description = Optional.ofNullable( description( rs ) );
-        LinkedHashSet<UUID> appTypeIds = appTypeIds( rs );
-        String url = url( rs );
-        return new App( id, name, title, description, appTypeIds, url );
+    public static Set<AppRole> appRoles( ResultSet rs ) throws SQLException, IOException {
+        return mapper.readValue( rs.getString( PostgresColumn.ROLES.getName() ), appRoleTypeRef );
```
[the remark this answers is not in the document]
Comment (makosblade) [reply src/main/java/com/openlattice/postgres/ResultSetAdapters.java]: Ignore me, this is JSONB

File: src/main/java/com/openlattice/postgres/ResultSetAdapters.java [hunk 5]
```diff
@@ -704,33 +707,39 @@ public static PrincipalSet principalSet( ResultSet rs ) throws SQLException {
...
     }
 
-    public static AppType appType( ResultSet rs ) throws SQLException {
-        UUID id = id( rs );
-        FullQualifiedName type = fqn( rs );
+    public static Map<UUID, AclKey> roles( ResultSet rs ) throws SQLException, IOException {
+        return mapper.readValue( rs.getString( PostgresColumn.ROLES.getName() ), rolesTypeRef );
```
[the remark this answers is not in the document]
Comment (makosblade) [reply src/main/java/com/openlattice/postgres/ResultSetAdapters.java]: Ignore me, this is JSONB

File: src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (makosblade) [reply src/main/java/com/openlattice/hazelcast/serializers/AppTypeSettingStreamSerializer.java]: Yeah I think I'm mixing it up with Kotlin

**Pull request merged**